### PR TITLE
added bindings for StartStopSilence, GapsDetector and Discontinuity Detector

### PR DESCRIPTION
**Pull request by ljoglar**
Here I added the code for the bindings for StartStopSilence, GapsDetector and Discontinuity Detector

Please, if you can, revise the code because I think it's not working properly... Otherwise I'll continue working on it when I can, no worries.

Thank you!

### CODE REVIEW

File: src/cpp/include/essentiamin.cpp [hunk 1]
```diff
@@ -577,6 +577,193 @@ std::vector<float> EssentiaMin::logMelBandsExtractor(std::vector<float>& signal,
 }
 
 
+std::vector<int> EssentiaMin::startStopSilence(std::vector<float>& signal) {
+
+    /////// PARAMS //////////////
+    int frameSize = 1024;
+    int hopSize = 512;
+    int staF = 0;
+    int stoF = 0;
```
Comment (pabloEntropia) [inline src/cpp/include/essentiamin.cpp]: These values could be parameters of the function

File: src/cpp/include/essentiamin.cpp [hunk 2]
```diff
@@ -577,6 +577,193 @@ std::vector<float> EssentiaMin::logMelBandsExtractor(std::vector<float>& signal,
...
+    delete w;
+    delete sss;
+
+    startStopFrames.push_back(staF);
+    startStopFrames.push_back(stoF);
+    return startStopFrames;
+}
+
+//Check https://essentia.upf.edu/documentation/reference/std_GapsDetector.html
+void EssentiaMin::gapsDetector(std::vector<float>& signal, std::vector<float>& starts, std::vector<float>& ends) {
```
Comment (pabloEntropia) [inline src/cpp/include/essentiamin.cpp]: while `startStopSilence` returns a vector with the results, `gapsDetector` and `discontinuityDetector` write them into separated vectors that are passed as a reference to the function. It'd be better to have consistent behavior. Maybe the easiest way would be to pass the result vector as a reference in `startStopSilence` too.

File: src/cpp/include/essentiamin.cpp [hunk 3]
```diff
@@ -577,6 +577,193 @@ std::vector<float> EssentiaMin::logMelBandsExtractor(std::vector<float>& signal,
...
+
+    /////// PARAMS //////////////
+    int frameSize = 1024;
+    int hopSize = 512;
```
Comment (pabloEntropia) [inline src/cpp/include/essentiamin.cpp]: Also hardcoded values here

File: src/cpp/include/essentiamin.cpp [hunk 4]
```diff
@@ -577,6 +577,193 @@ std::vector<float> EssentiaMin::logMelBandsExtractor(std::vector<float>& signal,
...
+                                    "frameSize", frameSize,
+                                    "hopSize", hopSize);
+
+    Algorithm* w  = factory.create("Windowing",
+                                   "type", "hann");
+
+    Algorithm* gd = factory.create("GapsDetector",
+                                   "frameSize", frameSize,
+                                   "hopSize", hopSize,
+                                   "silenceThreshold", -50);
```
Comment (pabloEntropia) [inline src/cpp/include/essentiamin.cpp]: Make this also a parameter?

File: src/cpp/include/essentiamin.cpp [hunk 5]
```diff
@@ -577,6 +577,193 @@ std::vector<float> EssentiaMin::logMelBandsExtractor(std::vector<float>& signal,
...
+    delete gd;
+}
+
+
+//Check https://essentia.upf.edu/documentation/reference/std_DiscontinuityDetector.html
+void EssentiaMin::discontinuityDetector(std::vector<float>& signal, std::vector<float>& discontinuityLocations, std::vector<float>& discontinuityAmplitudes) {
+
+    /////// PARAMS //////////////
+    int frameSize = 1024;
+    int hopSize = 512;
```
Comment (pabloEntropia) [inline src/cpp/include/essentiamin.cpp]: Same here

File: src/cpp/include/essentiamin.cpp [hunk 6]
```diff
@@ -577,6 +577,193 @@ std::vector<float> EssentiaMin::logMelBandsExtractor(std::vector<float>& signal,
...
+    while (true) {
+        // compute a frame
+        fc->compute();
+
+        // if it was the last one (ie: it was empty), then we're done.
+        if (!frame.size()) {
+            break;
+        }
+//        // if the frame is silent, just drop it and go on processing
+        if (isSilent(frame)){
```
Comment (pabloEntropia) [inline src/cpp/include/essentiamin.cpp]: Maybe we don't need to print anything on silent frames

**Comment by albincorreya** [comment]
Closing this since #11 adds generic bindings for all of the algorithms including the ones in this PR.

**Pull request closed without merging**